### PR TITLE
Fix: Bug on computeType with primitive types

### LINKED ISSUE #15: Extensions are not handled correctly when they are defined in the "type" section of a definition
example: Organization.partOf of Careplan.basedOn

These are references that HAVE an extension of type structuredefinition-hierarchy with a fixed valueBoolean. Pyrog display these attributes as extensions instead of references (fhir.ts --> attribute.ts, function "computeType" --> should add an extension to the attribute instead of modifying its type)

### PR DESCRIPTION
**Pull request by simonvadee**
Fix #15

### CODE REVIEW

File: tests/attribute.test.ts [hunk 1]
```diff
@@ -133,6 +149,16 @@ describe('Attribute', () => {
     `)
   })
 
+  it('computes the type correctly when dealing whith primitive types', () => {
+    const attribute = new Attribute(observationIdDefinition)
+    expect(attribute.types).toEqual(['string'])
+  })
+
+  it('computes the type correctly when dealing whith primitive types', () => {
```
[the remark this answers is not in the document]
Comment (Jasopaum) [reply tests/attribute.test.ts:157]: Also, this test doesn't have anything to do with the code you added, right? ^^

File: src/attribute.ts [hunk 2]
```diff
@@ -38,16 +38,21 @@ interface TypeElement {
...
 // - a profile (element.profile[0])
-// - an extension (element.extension[0].url)
 const computeType = (element: TypeElement): string => {
-  if (element.extension && element.extension.length > 0) {
-    return (
-      element.extension[0].valueUrl ||
-      element.extension[0].url.split('/').pop()!
+  const primitive =
+    element.extension &&
+    element.extension.find(
```
Comment (Jasopaum) [inline src/attribute.ts:47]: Any idea why this field is called extension in fhir?

File: tests/attribute.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonvadee) [reply tests/attribute.test.ts:157]: no, but the computeType function was not properly tested

[the remark this answers is not in the document]
Comment (simonvadee) [reply tests/attribute.test.ts:157]: updated the desc

File: src/attribute.ts (hunk 2 above)
Comment (simonvadee) [reply src/attribute.ts:47]: i believe it's when the attribute must have extensions with specific values

**Pull request merged**